pkg/graveler/committed/meta_range.go

```diff
@@ -24,6 +24,29 @@ type Iterator interface {
 	Close()
 }
 
+// DiffIterator iterates over all Range headers and values of a Diff, allowing seeking by entire
+// ranges.
+type DiffIterator interface {
+	// Next moves to look at the next value in the current Range, or a header for the next Range if the current Range is over and a next range exists.
+	Next() bool
+	// NextRange skips the current range
+	// possible only if we are currently inside a range
+	// the next value could be Range or Value
+	NextRange() bool
+	// Value returns a nil ValueRecord and a Range before starting a Range, or a Value and that Range when inside a Range.
+	// In contrast to Iterator, the DiffIterator might not have a current range
```

Comment:
Can you document _when_ there is no current range?  The code above seems to rely on that.